### PR TITLE
Initial implementation of Kotlin Formatter

### PR DESCRIPTION
**Pull request by wsutina**

### CODE REVIEW

File: gradle.properties [hunk 1]
```diff
@@ -0,0 +1,8 @@
+org.gradle.caching=true
+org.gradle.configuration-cache=true
+org.gradle.parallel=true
+
+dependency.analysis.print.build.health=true
+
+GROUP=xyz.block
+VERSION=1.0.0-SNAPSHOT
```
Comment (autonomousapps) [inline gradle.properties:8]: Just want to verify that you want your maven coordinates to be
```
xyz.block:kotlin-format:1.0.0
```
actually I think you don't! The `group` should be more fine-grained. I imagine there will be many libraries that want a group _prefixed with_ `xyz.block`. I think you might want something like this instead

```
xyz.block.kotlin-format:kotlin-format:1.0.0
```
Even though it looks redundant. It basically means "the `kotlin-format` artifact from the `xyz.block.kotlin-format` group". Note [that's what kotlin-editor does](https://github.com/cashapp/kotlin-editor/blob/main/gradle.properties#L14).

If you want more opinions about this, you could post in #opensource and tag Jessie Wilson and/or Jake Wharton. I had a [conversation](https://cash.slack.com/archives/C02FD8Z50/p1713289606948709) with Jessie a while ago about this there.

**Review by autonomousapps — APPROVED** [review]
LGTM, except for the `group`!

- [x] Check when group updated.

File: gradle.properties (hunk 1 above)
Comment (wsutina) [reply gradle.properties:8]: Good catch, I actually want it to be `xyz.block.kotlin-formatter:kotlin-formatter:1.0.0`

Comment (wsutina) [reply gradle.properties]: Fixed it. This is the maven coordinates now

```
➜  cat ~/.m2/repository/xyz/block/kotlin-formatter/kotlin-formatter/maven-metadata-local.xml 
<?xml version="1.0" encoding="UTF-8"?>
<metadata>
  <groupId>xyz.block.kotlin-formatter</groupId>
  <artifactId>kotlin-formatter</artifactId>
  <versioning>
    <latest>1.0.0-SNAPSHOT</latest>
    <versions>
      <version>1.0.0-SNAPSHOT</version>
    </versions>
    <lastUpdated>20241218001319</lastUpdated>
  </versioning>
</metadata>
```

**Pull request merged**